indico/modules/rb_new/client/js/common/bookings/OccurrenceActionsDropdown.jsx

```diff
@@ -0,0 +1,180 @@
+        const {dropdownOpen} = this.state;
+        const {top, left} = this.findPositioning();
+        if (dropdownOpen) {
+            this.setState({
+                dropdownOpen: false,
+                top,
+                left
+            });
+        } else {
+            this.setState({
+                dropdownOpen: true,
+                top,
+                left
+            });
+        }
+    };
+
+    renderRejectionForm = ({handleSubmit, hasValidationErrors, submitSucceeded, submitting, pristine}) => {
+        return (
+            <Form styleName="rejection-form" onSubmit={handleSubmit}>
+                <Field name="reason"
+                       component={ReduxFormField}
+                       as={TextArea}
+                       format={formatters.trim}
+                       placeholder={Translate.string('Provide the rejection reason')}
+                       rows={2}
+                       validate={v.required}
+                       disabled={submitting}
+                       required
+                       formatOnBlur />
+                <Button type="submit"
+                        disabled={submitting || pristine || hasValidationErrors || submitSucceeded}
+                        loading={submitting}
+                        floated="right"
+                        primary>
+                    <Translate>Reject</Translate>
+                </Button>
+            </Form>
+        );
+    };
+
+    render() {
+        const {activeConfirmation, dropdownOpen, top, left} = this.state;
+        const {booking: {canCancel, canReject}, date} = this.props;
+        const rejectionForm = (
+            <FinalForm onSubmit={(data) => this.changeOccurrenceState('reject', data)}
+                       render={this.renderRejectionForm} />
+        );
+        if (!canCancel && !canReject) {
+            return (null);
+        }
+        return (
+            <div styleName="actions-dropdown">
+                <Button styleName="dropdown-button" onClick={this.onClickButton}>
+                    <Button.Content>
+                        <Icon name="ellipsis horizontal" size="big" />
+                    </Button.Content>
+                </Button>
+                <Portal open={dropdownOpen} onClose={() => this.setState({dropdownOpen: false})}>
+                    <Dropdown icon={null}
+                              open
+                              style={{left: `${left}px`, position: 'fixed', top: `${top}px`, zIndex: 1000}}>
+                        <Dropdown.Menu direction="left">
+                            {canCancel && (
+                                <Dropdown.Item icon="times"
+                                               text="Cancel occurrence"
+                                               onClick={() => this.showConfirm('cancel')} />
+                            )}
+                            {canReject && (
+                                <Dropdown.Item icon="times"
```

Comment:
Would be nice to find a different icon for cancellation and rejection, but I'm no sure which one..

Reply:
Only admins in override mode can see both options; otherwise it's only one of them.